docs/ADR-65-avatar-system-for-renderer.md

```diff
@@ -0,0 +1,370 @@
+{
+    int lodIndex { get; }
+    void Bind(Renderer combinedAvatar);
+    void SetLodIndex(int lodIndex, bool inmediate = false);
+    void SetImpostorTexture(Texture2D texture);
+    void SetImpostorTint(Color color);
+}
+```
+
+`LOD` will also make use of the visibility handler (see below) to hide different parts of the avatar.
+
+### Visibility
+
+The final step of the loading process is to visibility handler, it’s not as easy as just turning on/off the avatar. Multiple systems have different reasons to hide or show an avatar and usually conflict with one another. An avatar can be hidden because it’s behind the camera, inside an `AvatarModifierArea` or because the max budget for avatars has been reached...
+
+To avoid these conflicts a visibility constrains system has been implemented.
+
+```csharp
+public interface IVisibility : IDisposable
+{
+    void Bind(Renderer combinedRenderer, List<Renderer> facialFeatures);
+
+    void AddGlobalConstrain(string key);
+    void RemoveGlobalConstrain(string key);
+
+    void AddCombinedRendererConstrain(string key);
+    void RemoveCombinedRendererConstrain(string key);
+
+    void AddFacialFeaturesConstrain(string key);
+    void RemoveFacialFeaturesConstrain(string key);
+}
+```
+
+The Avatar itself wont be rendered if any *global* or *CombinedRenderer* constrain exists.
+
+The *FacialFeatures* wont be rendered if any *global* or *FacialFeature* constrain exists.
+
+i.e. A constrain `own_player_invisible` will be added when toggling between 1st and 3rd person camera.
+
+## Tests suite
+
+The test suite is fairly simple compared to the complexity of the avatar system. Every dependency is injected in the constructor and based on an interface. This pattern called `DependencyInjection` allow isolation of every subsystem by mocking its dependencies using any of the mocking frameworks available (in our case `NSubstitute`).
+
+i.e.
+
+```csharp
+public class EmoteAnimationEquipperShould
+{
+    private EmoteAnimationEquipper equipper;
+    private IAnimator animator;
+
+    [SetUp]
+    public void SetUp()
+    {
+        animator = Substitute.For<IAnimator>();
+        equipper = new EmoteAnimationEquipper(animator);
+    }
+
+    [Test]
+    public void AssignReferencesOnConstruction()
+    {
+        Assert.AreEqual(animator, equipper.animator);
+        Assert.AreEqual(0, equipper.emotes.Count);
+    }
+}
+```
+
+## Benefits
+
+- Modularized: Every responsibility of the loading process is a separated module completly decoupled from the rest. It directly impact in the time required to find and fix bugs, maintain and scale each module.
+- Mockable: Test suite is more reliable, every aspect of the process can be tested without requiring visual tests.
+- Adaptability: Implementing new ways of loading and rendering avatars it’s easier. i.e. a LOD system with more than 3 levels can be implemented in Desktop without altering any other module.
+
+## Competition
+
+No alternatives have been consider.
```

Comment:
Danger

Reply:
the sentence `No alternatives have been consider.` is misleading and adds no context to the decision. 

I'd prefer you to add `The competition is the current system, which flaws are X, Y and Z. Adding to that, we are normalizing the implementation following best practices in the industry ...` or completely remove it.

Reply:
Fair enough! changing it in both ADRs